src/AzureExtension/DevBox/DevBoxProvider.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using System.Text.Json;
+using AzureExtension.Contracts;
+using DevHomeAzureExtension.DataModel;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Windows.DevHome.SDK;
+using Windows.Foundation;
+
+namespace AzureExtension.DevBox;
+
+public class DevBoxProvider : IComputeSystemProvider, IDisposable
+{
+    private readonly IHost _host;
+
+    public DevBoxProvider(IHost host)
+    {
+        _host = host;
+    }
+
+    string IComputeSystemProvider.DefaultComputeSystemProperties
+    {
+        get => throw new NotImplementedException();
+        set => throw new NotImplementedException();
+    }
+
+    public string DisplayName => "DevBox Provider";
+
+    public string Id => "182AF84F-D5E1-469C-9742-536EFEA94630";
+
+    public string Properties => throw new NotImplementedException();
+
+    public ComputeSystemProviderOperation SupportedOperations => throw new NotImplementedException();
+
+    private bool IsValid(JsonElement jsonElement)
+    {
+        return jsonElement.ValueKind != JsonValueKind.Undefined;
+    }
+
+    public IEnumerable<IComputeSystem> GetComputeSystemsAsync(IDeveloperId? developerId)

```

Comment:
This method is not marked async so will run synchronously.  Either
1. Remove `async` from the name
2. Add `async` to the declaration and add `await` to the method body.

Reply:
The implementation is async.